src/probestack.rs

```diff
@@ -43,18 +43,94 @@
 
 #![cfg(not(windows))] // Windows already has builtins to do this
 
+extern "C" {
+    pub fn __rust_probestack();
+}
+
+// A wrapper for our implementation of __rust_probestack, which allows us to
+// keep the assembly inline while controlling all CFI directives in the assembly
+// emitted for the function.
+//
+// This is the ELF version.
+#[cfg(all(
+    any(target_arch = "x86_64", target_arch = "x86"),
+    not(feature = "mangled-names"),
+    not(target_vendor = "apple")
+))]
+macro_rules! define_rust_probestack {
+    ($body: expr) => {
+        concat!(
+            "
+            // We are about to define a 'function within a function.' Because the
+            // compiler will have emitted a .cfi_startproc at the beginning of
+            // __rust_probestack_wrapper, we need .cfi_endproc before we can define
+            // the contents of __rust_probestack.
+            .cfi_endproc
+
+            .pushsection .text.__rust_probestack
+            .globl __rust_probestack
+            .type  __rust_probestack, @function
+        __rust_probestack:
```

Comment:
I believe on Mach-O the linker would be allowed to move this symbol and following content to another location, as it uses symbols to determine the start and end of every function.

Reply:
I think that would be fine. `__rust_probestack` is the function that gets called, not `__rust_probestack_wrapper`.